docket/core/sync_async.py

```diff
@@ -42,8 +53,10 @@ async def complete_callbacks_loop(self) -> None:
                 *[cb.complete() for cb in callbacks], return_exceptions=True
             )
 
-    def sync_call(self, awaitable: Awaitable[_T]) -> _T:
+    def sync_call(self, awaitable: Awaitable[_T]) -> _T | Exception:
```

Comment:
That's a valid type hint? 🧐

Reply:
Which one? _T is a TypeVar, Exception means an instance of an Exception.

```py
from typing import TypeVar
_T = TypeVar("_T")
def ret(x: _T) -> _T:
    return x

x = ret(1)
reveal_type(x)  # int

Reply:
It doesn't return an exception though? It raises one.

Reply:
Oh you're right, it used to return an Exception earlier but I changed that and forgot to fix the typehint.